content/assignments/techjobs-persistent/part-two/index.md

```diff
@@ -0,0 +1,105 @@
+---
+title: "Part Two"
+date: 2023-01-03T10:42:56-06:00
+draft: false
+weight: 3
+originalAuthor: John Woolbright # to be set by page creator
+originalAuthorGitHub: jwoolbright23 # to be set by page creator
+reviewer: Kimberly Horan # to be set by the page reviewer
+reviewerGitHub: codinglikeagirl42 # to be set by the page reviewer
+lastEditor: # update any time edits are made after review
+lastEditorGitHub: # update any time edits are made after review
+lastMod: # UPDATE ANY TIME CHANGES ARE MADE
+---
+
+## Persisting Employers and Skills
+
+You will need to have completed the [setup steps](https://education.launchcode.org/java-web-dev-curriculum/assignments/techjobs-persistent/part-one/#connect-a-database-to-a-spring-app) before starting this
+section.
+
+### AbstractEntity
+
+We've replaced the abstract class `JobField` with an even more abstracted class aptly named,
+`AbstractEntity`. This class holds the fields and methods that are common across the `Job` class
+and the classes it contains as fields.
+
+1. We will be creating tables for the subclasses that inherit from `AbstractEntity` but not a table for this parent class. Therefore, give `AbstractEntity` the `@MappedSuperclass` annotation.
+1. Since all of the subclasses of `AbstractEntity` will be entities themselves, add the `@Id` and `@GeneratedValue` annotations to the field `id`.
+1. Each subclass will also inherit the `name` field from `AbstractEntity`. Add appropriate validation annotations so that:
+    - a user cannot leave this field blank when creating an object.
+    - there are reasonable limitations on the size of the name string. Keep in mind that the name field will be shared across `Job`, `Employer`, and `Skill` classes. Some employer names might be longer than 50 characters.
+
+## Models
+
+In the last assignment, a `Job` object contained string fields for employer and core competency data. This employer
+and skill (formerly core competency) information about a particular job will now be stored in classes themselves.
+These items themselves will hold their own supplementary information.
+
+1. Open the `Employer` model class. In addition to the fields inherited from `AbstractEntity`, `Employer` should have a string field for `location`. Add the field for `location` with validation that ensures it is not empty and has a reasonable length. In addition, add public accessor methods to `Employer`.
+
+{{% notice blue Note "rocket" %}}
+For the purposes of this application, an employer can only have one location.
+{{% /notice %}}
```

Comment:
```suggestion
   {{% notice blue Note "rocket" %}}
   For the purposes of this application, an employer can only have one location.
   {{% /notice %}}
```

Reply:
This applies a strange font / box when tabbed in. Going to leave as is for now unless there is a fix I am unaware of.

Reply:
Ordered the numbers manually as a fix.